pkg/bufferdecoder/net_decoder.go

```diff
@@ -118,3 +120,80 @@ func (decoder *EbpfDecoder) DecodeNetDebugEvent(netDebugEvent *NetDebugEvent) er
 	decoder.cursor += int(netDebugEvent.GetSizeBytes())
 	return nil
 }
+
+// getDnsLayerFromBytes creates a packet from packetBytes and returns DNS layer
+func getDnsLayerFromBytes(packetBytes []byte) (*layers.DNS, error) {
+	packet := gopacket.NewPacket(packetBytes, layers.LayerTypeEthernet, gopacket.Default)
+	if packet == nil {
+		return nil, fmt.Errorf("couldn't parse the packet")
+	}
+	dnsLayer, ok := packet.Layer(layers.LayerTypeDNS).(*layers.DNS)
+	if !ok {
+		return nil, fmt.Errorf("couldn't find the DNS layer in packet")
+	}
+	return dnsLayer, nil
+}
+
+type DnsQueryData struct {
+	Query      string `json:"query"`
+	QueryType  string `json:"queryType"`
+	QueryClass string `json:"queryClass"`
+}
+
+// parseDnsQuestion parse dns request to DnsQueryData
+func parseDnsQuestion(question layers.DNSQuestion) DnsQueryData {
+	var request DnsQueryData
+	request.Query = string(question.Name)
+	request.QueryType = question.Type.String()
+	request.QueryClass = fmt.Sprint("", question.Class)
+	return request
+}
+
+// DecodeDnsQueryArray gets DNS layer from packet and parses DNS questions from it
+func (decoder *EbpfDecoder) DecodeDnsQueryArray(questions *[]DnsQueryData) error {
+	dnsLayer, err := getDnsLayerFromBytes(decoder.buffer[decoder.cursor:])
+	if err != nil {
+		return err
+	}
+
+	for _, question := range dnsLayer.Questions {
+		*questions = append(*questions, parseDnsQuestion(question))
+	}
+	return nil
+}
+
+type DnsAnswer struct {
+	Type   string `json:"answerType"`
+	Ttl    uint32 `json:"ttl"`
+	Answer string `json:"answer"`
+}
+
+type DnsResponseData struct {
+	QueryData DnsQueryData `json:"queryData"`
+	DnsAnswer []DnsAnswer  `json:"dnsAnswer"`
+}
+
+// DecodeDnsResponseData gets DNS layer from packet and parses DNS responses from it
+func (decoder *EbpfDecoder) DecodeDnsResponseData(responses *[]DnsResponseData) error {
+	dnsLayer, err := getDnsLayerFromBytes(decoder.buffer[decoder.cursor:])
+	if err != nil {
+		return err
+	}
+
+	for _, question := range dnsLayer.Questions {
+		response := DnsResponseData{}
+		response.QueryData = parseDnsQuestion(question)
+		if int(dnsLayer.ANCount) > 0 {
+			for _, answer := range dnsLayer.Answers {
+				ip := answer.IP.String()
+				if ip != "<nil>" {
+					response.DnsAnswer = append(response.DnsAnswer, DnsAnswer{Type: answer.Type.String(), Ttl: answer.TTL, Answer: ip})
+				} else {
+					response.DnsAnswer = append(response.DnsAnswer, DnsAnswer{Type: answer.Type.String(), Ttl: answer.TTL, Answer: string(answer.Data)})
```

Comment:
I believe you will have to treat different answers type instead of calling `string()` to the answer. Check this out:

```json
    {
      "name": "dns_response",
      "type": "[]bufferdecoder.DnsResponseData",
      "value": [
        {
          "queryData": {
            "query": "www.zip.net",
            "queryType": "AAAA",
            "queryClass": "IN"
          },
          "dnsAnswer": [
            {
              "answerType": "CNAME",
              "ttl": 290,
              "answer": "\bamazonas\u0003uol\u0003com\u0002br\u0000 <<<==== HERE"
            }
          ]
        }
      ]
    }
```